modules/@apostrophecms/express/index.js

```diff
@@ -215,6 +203,78 @@ module.exports = {
       },
       sessions: expressSession(self.getSessionOptions()),
       cookieParser: cookieParser(),
+      apiKeys(req, res, next) {
+        const key = req.query.apikey || req.query.apiKey || getAuthorizationApiKey();
+        let taskReq;
+        if (!key) {
+          return next();
+        }
+        if (_.has(self.options.apiKeys, key)) {
+          const info = self.options.apiKeys[key];
+          if (info.role === 'admin') {
+            taskReq = self.apos.task.getReq();
+          } else {
+            taskReq = self.apos.task.getAnonReq();
+          }
+          req.user = taskReq.user;
+          req.csrfExempt = true;
+          return next();
+        }
+        return res.status(403).send({ error: 'invalid api key' });
+
+        function getAuthorizationApiKey() {
+          const header = req.headers.authorization;
+          if (!header) {
+            return null;
+          }
+          const matches = header.match(/^ApiKey\s+(\S.*)$/i);
+          if (!matches) {
+            return null;
+          }
+          return matches[1];
+        }
+      },
+      beforeMiddleware(req, res, next) {
```

Comment:
What is the purpose of `beforeMiddleware`? Is it some kind of hook?

Reply:
Good catch it was testing